From a6c1e84fa71c5f7b63c28a76a569f536980f76c2 Mon Sep 17 00:00:00 2001
From: Erkai Ji <erkai.ji@nxp.com>
Date: Thu, 18 Dec 2025 19:17:03 +0800
Subject: [PATCH] LF-16845 Set default wayland initialization to swrast

Since Mesa marks wl_drm as legacy and relies on dmabuf to
retrieve DRM render node, this triggers warning on i.MX93.
Force software rendering here by changing LIBGL_ALWAYS_SOFTWARE
default value to true.

Upstream-Status: Inappropriate [i.MX specific]
Signed-off-by: Erkai Ji <erkai.ji@nxp.com>
---
 src/egl/main/eglapi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/egl/main/eglapi.c b/src/egl/main/eglapi.c
index 818c8edecc2..edded42306f 100644
--- a/src/egl/main/eglapi.c
+++ b/src/egl/main/eglapi.c
@@ -677,7 +677,7 @@ eglInitialize(EGLDisplay dpy, EGLint *major, EGLint *minor)
    if (!disp->Initialized) {
       /* set options */
       disp->Options.ForceSoftware =
-         debug_get_bool_option("LIBGL_ALWAYS_SOFTWARE", false);
+         debug_get_bool_option("LIBGL_ALWAYS_SOFTWARE", true);
       if (disp->Options.ForceSoftware)
          _eglLog(_EGL_DEBUG,
                  "Found 'LIBGL_ALWAYS_SOFTWARE' set, will use a CPU renderer");
-- 
2.34.1

